modules/objdetect/include/opencv2/objdetect/face.hpp

```diff
@@ -0,0 +1,105 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef _OPENCV_DNN_FACE_CORE_HPP_
+#define _OPENCV_DNN_FACE_CORE_HPP_
+
+#include <opencv2/core.hpp>
+
+/** @defgroup dnn_face DNN-based face detection and recognition
+ */
+
+namespace cv
+{
+
+/** @brief DNN-based face detector, model download link: https://github.com/ShiqiYu/libfacedetection.train/tree/master/tasks/task1/onnx.
+ */
+class CV_EXPORTS_W FaceDetector
+{
+public:
+    virtual ~FaceDetector() {};
```

Comment:
What is about reusing/adopting of "High Level DNN API"?

Reply:
The model we used can detect faces with landmarks, so existing object detection APIs do not fit our need and we create a new one.

Reply:
"DNN High Level  API" is not about object detection at all. It is mostly about configuration of underlying DNN networks execution in some generic way. At least to configure execution backend/target to inference network on GPU.